### DISCUSSION TITLE
Error running input file in CFAST gfortran build

### DISCUSSION THREAD

**Discussion by fttoinches** [firemodels/cfast#2100] [Q&A] [answered]
_I followed the instructions outlined in the wiki to build cfast from the makefile on windows intel 64 using gnu, which was successful with no errors. I used the latest repository for this build (no changes in the source files were made). I created an input file and ran a simulation (see input file below) in Cedit 7.7.4 with no problems._

_Input file:_

&HEAD VERSION = 7700, TITLE = 'CFAST Simulation' /
!! Scenario Configuration 
&TIME SIMULATION = 5400 PRINT = 60 SMOKEVIEW = 15 SPREADSHEET = 15 / 
&INIT PRESSURE = 101325 RELATIVE_HUMIDITY = 50 INTERIOR_TEMPERATURE = 20 EXTERIOR_TEMPERATURE = 20 /
 
!! Material Properties 
&MATL ID = 'NM 1' MATERIAL = 'New Material 1', 
      CONDUCTIVITY = 0.033 DENSITY = 160 SPECIFIC_HEAT = 1.03, THICKNESS = 0.075 EMISSIVITY = 0.9 /
&MATL ID = 'NM 2' MATERIAL = 'New Material 2', 
      CONDUCTIVITY = 0.1 DENSITY = 500 SPECIFIC_HEAT = 1.76, THICKNESS = 0.016 EMISSIVITY = 0.9 /
!! Compartments 
&COMP ID = 'Comp 1'
      DEPTH = 1 HEIGHT = 1.3 WIDTH = 1.5
      CEILING_MATL_ID = 'NM 2' CEILING_THICKNESS = 0.3 WALL_MATL_ID = 'NM 1' WALL_THICKNESS = 0.3 FLOOR_MATL_ID = 'NM 2'
      ORIGIN = 0, 0, 0 GRID = 50, 50, 50 /
!! Wall Vents
&VENT TYPE = 'WALL' ID = '1' COMP_IDS = 'Comp 1' 'OUTSIDE' , BOTTOM = 0 HEIGHT = 1 WIDTH = 0.5
      FACE = 'FRONT'  OFFSET = 0.5 /
!! Fires 
&FIRE ID = 'fire1'  COMP_ID = 'Comp 1', FIRE_ID = 'New Fire 6'  LOCATION = 0.25, 0.25 / 
&CHEM ID = 'New Fire 6' CARBON = 1 CHLORINE = 0 HYDROGEN = 4 NITROGEN = 0 OXYGEN = 0 HEAT_OF_COMBUSTION = 50000 RADIATIVE_FRACTION = 0.35 / 
&TABL ID = 'New Fire 6' LABELS = 'TIME', 'HRR' , 'HEIGHT' , 'AREA' , 'CO_YIELD' , 'SOOT_YIELD' , 'HCN_YIELD' , 'HCL_YIELD' , 'TRACE_YIELD'  /
&TABL ID = 'New Fire 6', DATA = 0, 0.1, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 1, 5, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 1200, 20, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 1201, 20, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 1525, 20, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 1526, 5, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 4000, 5, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 4001, 5, 0, 0.09, 0, 0, 0, 0, 0 /
&TABL ID = 'New Fire 6', DATA = 5400, 0, 0, 0.09, 0, 0, 0, 0, 0 /
&TAIL /

_To test if my build works, I copied the .in file into the directory of the build and ran it, I get two types of errors depending on the HRR I specify in the first time step. If I specify zero HRR at time zero I get the following error:_

***Error, at t (=r1) and stepsize h (=r2) the initial yprime could not be computed
ierror,r1,r2 =  685    0.0000D+00    0.0000D+00
Solution component with the greatest error is
upper layer temp in room  1
***Error, dassl - idid = -12
Note: The following floating-point exceptions are signalling: IEEE_UNDERFLOW_FLAG IEEE_DENORMAL

_If I set the HRR at say 0.1 at time = 0 I get the following error:_

Program received signal SIGFPE: Floating-point exception - erroneous arithmetic operation.
 
Backtrace for this error:
#0  0x9649bbda
#1  0x9648dc63
#2  0x9646ecf6
#3  0xca27b247
#4  0xcae7441e
#5  0xcadee465
#6  0xcae7340d
#7  0x963f3606
#8  0x963f4dd6
#9  0x9643566b
#10  0x9640362c
#11  0x96437ec8
#12  0x96462fac
#13  0x964b9549
#14  0x963e13ad
#15  0x963e14e5
#16  0xca10257c
#17  0xcae2aa57
#18  0xffffffff

_I have tried specifying MAX_TIME_STEP = 0.01 on the &misc line but to no avail._

_Could you please assist me with this? Thank you in advance._

**Comment by preneke** [comment]
I ran your case in the CFAST 7.7.4, both versions of your case and they run fine. There appears to be a problem with you code. Unfortunately it is impossible for me to debug your code via messages. All I can say is that I have never seen that error before. Sorry I can't be of more help. Good luck.

**Reply by fttoinches** [reply]
Thanks! I appreciate you taking the time to take a look at this.

**Reply by rpeacoc** [reply]
Paul, you may want to create an issue with this test case to see why it's failing with gnu with optimized options.

**Comment by mcgratta** [comment, marked as answer]
The `IEEE_UNDERFLOW` message probably means that the compiler does not automatically assume that 1e-99, say, can be considered 0. The gnu compiler can be fussy about such things and there is probably a flag to tell it to relax. As for the floating point exception, that might be a divide by zero or a square root of a very small negative number. Could you recompile using the `gnu_win_64_db` build target and run the case again. That might provide us with a line number indicating where the problem is.

**Reply by fttoinches** [reply]
Recompiling using `gnu_win_64_db` seemed to have solved the problem. Both cases now run fine without any errors, thank you!

**Comment by mcgratta** [comment]
OK, but that still leaves open the issue of why the job fails when compiled with full optimization.

**Comment by preneke** [comment]
Yea, you were right. Thanks.
